app/assets/src/components/views/discovery/DiscoveryDataLayer.js

```diff
@@ -0,0 +1,104 @@
+import { range } from "lodash/fp";
+import { getDiscoverySamples } from "./discovery_api";
+
+export default class DiscoveryDataLayer {
+  constructor(domain) {
+    this.domain = domain;
+
+    this.data = {
+      samples: this.newObjectDb(),
+    };
+
+    this.apiFunctions = {
+      samples: this.fetchSamples,
+    };
+  }
+
+  newObjectDb = () => ({
+    entries: {},
+    order: null,
```

Comment:
`order` seems a little vague (at first I was expecting it to be a sorting function) ?

Reply:
orderedIds?

Reply:
yeah that sounds good